src/Traces.Core/Extensions/Files/CreateTraceFileDtoExtensions.cs

```diff
@@ -0,0 +1,62 @@
+using System;
+using System.Collections.Generic;
+using Traces.Common;
+using Traces.Common.Constants;
+using Traces.Common.Utils;
+using Traces.Core.Models.Files;
+using Traces.Data.Entities;
+
+namespace Traces.Core.Extensions.Files
+{
+    public static class CreateTraceFileDtoExtensions
+    {
+        public static bool IsValid(this CreateTraceFileDto createTraceFileDto)
+        {
+            if (string.IsNullOrWhiteSpace(createTraceFileDto.Name) ||
+                string.IsNullOrWhiteSpace(createTraceFileDto.MimeType) ||
+                createTraceFileDto.Size <= 0 ||
+                createTraceFileDto.Size > AppConstants.MaxFileSizeInBytes ||
+                createTraceFileDto.Data == null
+            )
+            {
+                return false;
+            }
+
+            return true;
+        }
+
+        public static bool IsValid(this List<CreateTraceFileDto> dtos)
+        {
+            foreach (var createTraceFileDto in dtos)
+            {
+                if (!createTraceFileDto.IsValid())
+                {
+                    return false;
+                }
+            }
+
+            return true;
+        }
+
+        public static TraceFile ToTraceFile(this CreateTraceFileDto dto, IRequestContext requestContext)
+        {
+            Check.NotNull(dto, nameof(dto));
+            Check.NotNull(requestContext, nameof(requestContext));
+
+            var tenantId = requestContext.TenantId;
+            var fileGuid = Guid.NewGuid();
+            var currentSubjectId = requestContext.SubjectId;
+            var path = $"files/{tenantId}/{fileGuid}/{dto.Name}";
```

Comment:
```suggestion
        public static TraceFile ToTraceFile(this CreateTraceFileDto dto)
        {
            Check.NotNull(dto, nameof(dto));
```
This function should only be for mapping from one type to another.

This should be done in the function that calls this, and passed as properties in the CreateTraceFileDto